File: tnzhou/334.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        if (nums.size() < 3)
            return false;
        int smallestIndex = 0;
        int midNumIndex = -1;
        for (int i = 1; i < nums.size(); i ++) {
            // Bigger the second large number, found the largest one.
            if (midNumIndex != -1 && nums[i] > nums[midNumIndex])
                return true;
            // Bigger number found after the smallest, update the index.
            if (nums[i] > nums[smallestIndex] && (midNumIndex == -1 || nums[i] < nums[midNumIndex]))
                midNumIndex = i;
            // No matter to change smallest even it's behind the original second large number.
            // It's alreay a smaller number before the second large number.
            if (nums[i] < nums[smallestIndex])
                smallestIndex = i;
        }
        return false;
    }
};

File: zemiao/334.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        a,b = None,None
        for x in nums:
          if a==None or x<=a:
            a = x
            continue
          if b==None or x<=b:
            b = x
            continue
          return True
        return False